kafka-rest/src/main/java/io/confluent/kafkarest/resources/v3/V3ResourcesModule.java

```diff
@@ -30,5 +37,40 @@ protected void configure() {
     bindAsContract(ChunkedOutputFactory.class);
     bindAsContract(StreamingResponseFactory.class);
     bind(Clock.systemUTC()).to(Clock.class);
+    bindFactory(ExecutorServiceFactory.class, Singleton.class)
+        .to(ExecutorService.class)
+        .in(Singleton.class);
+  }
+
+  private static final class ExecutorServiceFactory implements Factory<ExecutorService> {
+
+    private final int produceExecutorThreadPoolSize;
+    private volatile ExecutorService executorService;
+
+    @Inject
+    ExecutorServiceFactory(
+        @ProduceExecutorThreadPoolSizeConfig Integer produceExecutorThreadPoolSize) {
+      this.produceExecutorThreadPoolSize = requireNonNull(produceExecutorThreadPoolSize);
+    }
+
+    @Override
+    public ExecutorService provide() {
+      ExecutorService localExecutorService = executorService;
+      if (localExecutorService == null) {
+        synchronized (this) {
+          localExecutorService = executorService;
+          if (localExecutorService == null) {
+            executorService =
+                localExecutorService = Executors.newFixedThreadPool(produceExecutorThreadPoolSize);
```

Comment:
Just return this directly. HK2 will save the instance for you int the `Singleton` scope.